source/initial-dns-seedlist-discovery/initial-dns-seedlist-discovery.md

```diff
@@ -81,31 +81,51 @@ parse error and MUST NOT do DNS resolution or contact hosts.
-name, and a TLD. Drivers MUST raise an error and MUST NOT contact the DNS server to obtain SRV (or TXT records) if the
-full URI does not consist of at least three parts.
+Node:
 
 If `mongodb+srv` is used, a driver MUST implicitly also enable TLS. Clients can turn this off by passing `tls=false` in
 either the Connection String, or options passed in as parameters in code to the MongoClient constructor (or equivalent
 API for each driver), but not through a TXT record (discussed in a later section).
 
+#### Terminology
+
+For the purposes of this document, `{hostname}` will be divided using the following terminology.
+
+If an SRV `{hostname}` has:
+
+- 3 or more `.` separated parts:
+  - the left-most part is the `{subdomain}`
+  - the remaining portion is the `{domain}`
+  - examples:
+    - `{cluster_1.tests.mongodb.co.uk}`
+    - `{hosts_34.example.com}`
+- have 1 or 2 `.` separated part(s):
+  - the `{hostname}` is equivalent to the `{domain}`
+  - examples:
+    - `{localhost}`
+    - `{mongodb.local}`
+
 #### Querying DNS
 
-In this preprocessing step, the driver will query the DNS server for SRV records on `{hostname}.{domainname}`, prefixed
-with the SRV service name and protocol. The SRV service name is provided in the `srvServiceName` URI option and defaults
-to `mongodb`. The protocol is always `tcp`. After prefixing, the URI should look like:
-`_{srvServiceName}._tcp.{hostname}.{domainname}`. This DNS query is expected to respond with one or more SRV records.
+In this preprocessing step, the driver will query the DNS server for SRV records on the hostname, prefixed with the SRV
+service name and protocol. The SRV service name is provided in the `srvServiceName` URI option and defaults to
+`mongodb`. The protocol is always `tcp`. After prefixing, the URI should look like: `_{srvServiceName}._tcp.{hostname}`.
+This DNS query is expected to respond with one or more SRV records.
 
 The priority and weight fields in returned SRV records MUST be ignored.
 
 If the DNS result returns no SRV records, or no records at all, or a DNS error happens, an error MUST be raised
 indicating that the URI could not be used to find hostnames. The error SHALL include the reason why they could not be
 found.
 
-A driver MUST verify that the host names returned through SRV records have the same parent `{domainname}`. Drivers MUST
-raise an error and MUST NOT initiate a connection to any returned host name which does not share the same
+A driver MUST verify that the host names returned through SRV records end with the original SRV's `{domainname}`.
+Drivers MUST raise an error and MUST NOT initiate a connection to any returned host name which does not share the same
 `{domainname}`.
 
+In the case that the SRV record has less than three `.` separated parts, the returned address MUST NOT be identical to
+the original `{hostname}`. The next major version should the host names returned through all SRVs require to end with
```

Comment:
Can you make "all SRVs" more specific about what we want to disallow? "drivers must no longer permit __ for any number of parts" something along those lines,

When it becomes the new normal for drivers to assert this we will want to understand what the old way was through the prohibition of it.